### DISCUSSION TITLE
How to create xml that one body drag another box with cable.

### DISCUSSION THREAD

**Discussion by HuanMinpopcorn** [google-deepmind/mujoco#2308] [Asking for Help]
### Intro

Hi!

I am a graduate student, I use MuJoCo for my research on leg robot.

### My setup

mujoco 3.2.5
python3 

### My question

I'm tryinig to connect the rope between two boxes and also try to driving one box while drag another box. 

### Minimal model and/or code that explain my question

<mujoco model="RopeBetweenBoxes">
  <include file="scene.xml"/>
  <extension>
    <plugin plugin="mujoco.elasticity.cable"/>
  </extension>
  <statistic center="0 0 0" extent="1"/>
  <visual>
    <global elevation="-30"/>
  </visual>

  <size memory="2M"/>

  <worldbody>
    
    <body name="box1" pos="-.15 0 0.1">
      <geom type="box" size="0.1 0.1 0.1" mass="1.0" rgba="0 0 1 1"/>
      <site name="box1_site" pos="0 0 0"/>
      <joint name="box1_slide" type="slide" axis="1 0 0" damping="0.01"/>
    </body>

    
    <body name="box2" pos="0.85 0 0.1">
      <geom type="box" size="0.1 0.1 0.1" mass=".2" rgba="1 0 0 1"/>
      <site name="box2_site" pos="0 0 0"/>
      
    </body>

    <composite type="cable" curve="s" count="41 1 1" size="1" offset="0 0 0.1" initial="none">
      <plugin plugin="mujoco.elasticity.cable">
      
        <config key="twist" value="1e7"/>
        <config key="bend" value="4e6"/>
        <config key="vmax" value="0.05"/>
      </plugin>
      <joint kind="main" damping=".015"/>
      <geom type="capsule" size=".005" rgba=".8 .2 .1 1" condim="1"/>
    </composite>
  </worldbody>
  <equality>
    <weld name="left_boundary" body1="B_first" body2="box1" anchor="-.025 0 0"/>
    <weld name="right_boundary" body1="B_last" body2="box2" anchor=".025 0 0.1"/>
  </equality>
  <contact>
    <exclude body1="B_first" body2="box1"/>
    <exclude body1="B_last" body2="box2"/>
  </contact>
  <actuator>
    
    <motor joint="box1_slide" gear="1" ctrlrange="-10 10"/>
  </actuator>
</mujoco>

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by HuanMinpopcorn** [comment]
the problem I have now, the cable is not attached to the second box. The first box doesn't move while been added the slide joint.

**Comment by yuvaltassa** [comment]
This was asked many times before. Please search the issues.
